### PR TITLE
Update UPDATING.md

### PR DESCRIPTION
**Pull request by btangmu**
- Add instructions for setting VERSION in cldr-config.sh, uploading zipfiles to release page

- Add a few notes for clarification

- Update copyright year

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -34,14 +34,18 @@ Make sure your `cldr-json` directory is otherwise clean (`git status`)
 
 ### Publishing
 
-1. Run the script `cldr-generate-zip.sh` to generate zipfiles under `dist/` - these are suitable for uploading to a release page
-2. create a release tag such as `43.0.0` in this repository. or `43.0.0-ALPHA2`.  Create a GitHub release, use other [releases](https://github.com/unicode-org/cldr-json/releases) as a guide.
-3. npm packages can be updated as well. Each sub-subdirectory of `cldr-json/cldr-json` is a separate npm package. The following script will preview
+1. Run the script `cldr-generate-zip.sh` to generate zipfiles under `dist/`, then rename them by inserting the release tag between the two hyphens; for example, rename `cldr--json-full.zip` to `cldr-44.0.0-ALPHA2-json-full.zip`.
```
Comment (srl295) [inline UPDATING.md:37]: Not correct.

`VERSION` should be set in your local config .sh file to the correct version such as `44.0.0-ALPHA2` and then the zip file will be created correctly.

IN fact it looks like the npms published were published as `44.0.0` which is final data. This isn't correct and we'll have to deprecate them or unpublished them and have the final version be 44.0.1 or some such.

**Review by srl295 — CHANGES_REQUESTED** [review]

**Comment by srl295** [comment]
I added a comment on #53 about production problems there

File: UPDATING.md (hunk 1 above)
Comment (btangmu) [reply UPDATING.md:37]: "published as 44.0.0 which is final data" -- I did include the "-ALPHA2" in the names; sorry I'm not quite getting it -- sounds like UPDATING.md lacks instructions about setting VERSION? Or did I miss that?

**Comment by srl295** [comment]
You didn’t set VERSION to 44.0.0-BETA3 or whatever the tag is.  That caused all of the cldr-json packages to have “version: “44.0.0”” which is now published on npm, which isn’t reversible.  https://www.npmjs.com/package/cldr-core?activeTab=versions 

Maybe a better tool default if VERSION isn’t set is to fail rather than auto detect.  ANd if it does detect, it could make the version “44.0.0-UNKNOWN” or something that would prevent this problem.

**Comment by btangmu** [comment]
```
$ cat cldr-config.sh
#!/bin/sh

# Don't modify this file for temporary changes.
# create an executable ./local-config.sh and it will be used instead.

# where CLDR is
CLDR_DIR=../cldr

# output of generated JSON
OUT=./cldr-json

# input data, assumed to be in a parallel repo
# <https://github.com/unicode-org/cldr-staging.git>
INDATA=../cldr-staging/production

# output of .zip files
DIST=./dist

# which types to generate
TYPES=all

# Only these files
MATCH='.*'

# what draft status to consider
DRAFTSTATUS="contributed"

# override the version number of CLDR
#VERSION="39.0.0"

EXTRA_JSON_OPTS=""
```

VERSION is commented out -- @srl295 it does sound like a good idea for scripts that depend on it to fail rather than auto detect. I'm not sure what "auto detect" currently means -- we actually just get an empty string if VERSION isn't set, as in "cldr--json-full.zip" instead of "cldr-44.0.0-ALPHA2-json-full.zip". Auto detect would somehow get VERSION from the cldr folder? But cldr is also still on alpha...

**Comment by srl295** [comment]
> ```
> $ cat cldr-config.sh
> #!/bin/sh
> 
> # Don't modify this file for temporary changes.
> # create an executable ./local-config.sh and it will be used instead.
> 
> # where CLDR is
> CLDR_DIR=../cldr
> 
> # output of generated JSON
> OUT=./cldr-json
> 
> # input data, assumed to be in a parallel repo
> # <https://github.com/unicode-org/cldr-staging.git>
> INDATA=../cldr-staging/production
> 
> # output of .zip files
> DIST=./dist
> 
> # which types to generate
> TYPES=all
> 
> # Only these files
> MATCH='.*'
> 
> # what draft status to consider
> DRAFTSTATUS="contributed"
> 
> # override the version number of CLDR
> #VERSION="39.0.0"
> 
> EXTRA_JSON_OPTS=""
> ```
> 
> 
>     
>   
> 
> VERSION is commented out -- @srl295 it does sound like a good idea for scripts that depend on it to fail rather than auto detect. I'm not sure what "auto detect" currently means -- we actually just get an empty string if VERSION isn't set, as in "cldr--json-full.zip" instead of "[cldr-44](https://unicode-org.atlassian.net/browse/CLDR-44).0.0-ALPHA2-json-full.zip". Auto detect would somehow get VERSION from the cldr folder? But cldr is also still on alpha...

You can create an executable `./local-config.sh` file containing `VERSION=44.0.0-ALPHA2`

Auto-detect means that when you generate json, it reads `CLDRFile.GEN_VERSION` which is 44.  But yes, I think the scripts should just require an explicit version every time.

**Comment by btangmu** [comment]
I've updated UPDATING.md again for the PR

**Comment by btangmu** [comment]
is it better to create local-config.sh than to edit cldr-config.sh?

**Comment by btangmu** [comment]
> is it better to create local-config.sh than to edit cldr-config.sh?

I see, that's for "temporary changes" -- at some point though cldr-config.sh itself should be updated for current VERSION. And yes, falling back on CLDRFile.GEN_VERSION seems more dangerous than helpful

**Comment by srl295** [comment]
> is it better to create local-config.sh than to edit cldr-config.sh?

Yes, that's what the note is trying to say

File: UPDATING.md [hunk 2]
```diff
@@ -34,14 +34,19 @@ Make sure your `cldr-json` directory is otherwise clean (`git status`)
 
 ### Publishing
 
-1. Run the script `cldr-generate-zip.sh` to generate zipfiles under `dist/` - these are suitable for uploading to a release page
-2. create a release tag such as `43.0.0` in this repository. or `43.0.0-ALPHA2`.  Create a GitHub release, use other [releases](https://github.com/unicode-org/cldr-json/releases) as a guide.
-3. npm packages can be updated as well. Each sub-subdirectory of `cldr-json/cldr-json` is a separate npm package. The following script will preview
+1. Edit the file `cldr-config.sh` and set `VERSION` appropriately; for example, `VERSION="44.0.0-ALPHA2"`.
```
Comment (srl295) [inline UPDATING.md:37]:
```suggestion
1. Create the file `local-config.sh`, make it executable, and set `VERSION` appropriately; for example, `VERSION="44.0.0-ALPHA2"`.
```

**Review by srl295 — APPROVED** [review]

**Comment by btangmu** [comment]
@srl295 when and why should cldr-identify.sh and cldr-jsonlint.sh be run?

**Pull request merged**